Apsitvarkom.DataAccess/ILocationDTORepository.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Apsitvarkom.Models;
+using Apsitvarkom.Models.DTO;
+
+namespace Apsitvarkom.DataAccess;
+
+/// <summary>
+/// Interface for <see cref="T" /> class data handling, where <see cref="T"/> derives from <see cref="LocationDTO"/>.
+/// </summary>
+public interface ILocationDTORepository<T> : IEntityDTORepository<T> where T : LocationDTO
```

Comment:
I wonder if we're planning on having multiple types derived from `LocationDTO`. We now only have `PollutedLocationDTO` and I didn't think we'll be adding more types derived from `LocationDTO`, so my initial thought was to have this interface non generic:

`public interface IPollutedLocationDTORepository : IEntityDTORepository<PollutedLocationDTO>`

Just because you *could* have it generic, doesn't mean that you *need* it to have generic. Although we can keep it as it is right now if you think we could be adding more types deriving from `LocationDTO` in the future.

Reply:
I wasn't sure on how we would handle those `Event` type classes - whether we would derive it from PollutedLocationDTO, have the related PollutedLocationDTO as a property inside of it or have some separate class without much direct relation to the PollutedLocationDTO (hence probably needing to derive it from LocationDTO). Maybe let's keep it as is until we figure this one out and could change it if needed afterwards?

Reply:
Sure, we can keep it as is for now and change it if needed. 

> I wasn't sure on how we would handle those Event type classes

Composition probably (that is the "have the related PollutedLocationDTO as a property inside of it" option). In a database we will probably store a foreign key to the polluted location ID.